### PR TITLE
Merge universes without coordinates Fix 2470

### LINKED ISSUE #2470: mda.Merge for Universes without coordinates
**Expected behavior**

I can make a new Universe with mda.Merge from my old Universe without positions.

**Actual behavior**

I get an AttributeError because mda.Merge does not wrap the `coords` in a try/except clause. MDAnalysis can make Universes without coordinates, so when the `args` don't have `positions`, the `coords[None, :, :]` argument to `Universe()` can be left out.

https://github.com/MDAnalysis/mdanalysis/blob/b5cb49fc06c05d4038f75ba98c66dc30f856d303/package/MDAnalysis/core/universe.py#L1534-L1537

**Code to reproduce the behavior**

Show us how to reproduce the failiure. If you can, use trajectory files from the test data.

``` python
>>> import MDAnalysis as mda
>>> from MDAnalysis.tests.datafiles import PSF
>>> u = mda.Universe(PSF)
>>> subset = mda.Merge(u.atoms[:10])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/universe.py", line 1535, in Merge
    coords = np.vstack([a.positions for a in args])
  File "/Users/lily/pydev/mdanalysis/package/MDAnalysis/core/universe.py", line 1535, in <listcomp>
    coords = np.vstack([a.positions for a in args])
  File "mdanalysis/package/MDAnalysis/core/groups.py", line 2294, in __getattr__
    cls=self.__class__.__name__, attr=attr))
AttributeError: AtomGroup has no attribute positions
```

**Currently version of MDAnalysis**

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 0.20.2-dev
- Which version of Python (`python -V`)? 3.6
- Which operating system? MacOS

### PR DESCRIPTION
**Pull request by AnshulAngaria**
Fixes #2470

Changes made in this Pull Request:
 - created and populated universes under try/except clause so that when args don't have positions the coords[None, :, :] argument to Universe() can be left out.

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/universe.py [hunk 1]
```diff
@@ -1498,8 +1498,11 @@ def Merge(*args):
...
     # Create and populate a universe
-    coords = np.vstack([a.positions for a in args])
-    u = Universe(top, coords[None, :, :],
+    try:
+        coords = np.vstack([a.positions for a in args])
+        u = Universe(top, coords[None, :, :],
                  format=MDAnalysis.coordinates.memory.MemoryReader)
-
+    except:
+        u = Universe(top, format=MDAnalysis.coordinates.memory.MemoryReader)
```
Comment (richardjgowers) [inline package/MDAnalysis/core/universe.py]: You don't need to specify `format=` here as there are no coordinates

Comment (AnshulAngaria) [reply package/MDAnalysis/core/universe.py]: I have added tests and changelog entry and removed format=.
Thanks for the help.

File: package/CHANGELOG [hunk 2]
```diff
@@ -20,6 +20,7 @@ mm/dd/yy richardjgowers, kain88-de, lilyminium, p-j-smith, bdice, joaomcteixeira
   * 0.21.0
 
 Fixes
+  * Create universes under try/except clause.(Issue #2470)(PR #2580)
```
Comment (orbeckst) [inline package/CHANGELOG]: Also add your GitHub to the authors list

File: package/MDAnalysis/core/universe.py [hunk 3]
```diff
@@ -1498,8 +1498,11 @@ def Merge(*args):
...
 
     # Create and populate a universe
-    coords = np.vstack([a.positions for a in args])
-    u = Universe(top, coords[None, :, :],
+    try:
+        coords = np.vstack([a.positions for a in args])
+        u = Universe(top, coords[None, :, :],
                  format=MDAnalysis.coordinates.memory.MemoryReader)
-
+    except:
```
Comment (orbeckst) [inline package/MDAnalysis/core/universe.py]: Explicitly list the exceptions on which the except clause triggers. Blanket exception catching will lead to very hard to find bugs.

File: package/CHANGELOG (hunk 2 above)
Comment (richardjgowers) [reply package/CHANGELOG]: The entries in CHANGELOG are meant to indicate to users what has been fixed, ie here Merging Universes without coordinates.  This entry details what you did to fix it, but it's not useful information here

Comment (AnshulAngaria) [reply package/CHANGELOG]: I have updated CHANGELOG. Will take care of it next time.

**Comment by richardjgowers** [comment]
cool thanks for the fix @AnshulAngaria !

**Comment by AnshulAngaria** [comment]
@richardjgowers Thank you! Excited for doing more contributions!

**Pull request merged**